### PR TITLE
PIMS-1546 Disable Edit Button on User Detail Page for Non-Admins

### PR DESCRIPTION
**Pull request by TaylorFries**
## 🎯 Summary

  
[PIMS-1546: ](https://citz-imb.atlassian.net/browse/PIMS-1546)

Added line to get users keycloak role, if the user is an admin the edit button is enabled, if the user is not an admin the edit button is disabled. 

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: react-app/src/components/users/UserDetail.tsx [hunk 1]
```diff
@@ -96,6 +98,11 @@ const UserDetail = ({ onClose }: IUserDetail) => {
     mode: 'onBlur',
   });
 
+  let canEdit = false;
+  if (user.keycloak.hasRoles([Roles.ADMIN])) {
+    canEdit = true;
+  }
+
```
Comment (dbarkowsky) [inline react-app/src/components/users/UserDetail.tsx:105]: Could probably condense this.

```suggestion
  const canEdit = user.keycloak.hasRoles([Roles.ADMIN]);
```

File: react-app/src/components/users/UserDetail.tsx [hunk 2]
```diff
@@ -27,8 +28,9 @@ interface UserProfile extends User {
 
 const UserDetail = ({ onClose }: IUserDetail) => {
   const { id } = useParams();
-  const { pimsUser } = useContext(AuthContext);
+  const user = useContext(AuthContext);
   const api = usePimsApi();
+  const pimsUser = { user };
```
Comment (dbarkowsky) [inline react-app/src/components/users/UserDetail.tsx:33]: This change is creating an extra layer to this object, which is why you needed the change on line 139. 
useContext will give back something like { pimsUser, keycloakUser }, so the original used destructuring to get pimsUser directly.
Would recommend keeping the original, but just getting keycloakUser through destructuring as well so you can use it for the hasRoles function below.

Comment (TaylorFries) [reply react-app/src/components/users/UserDetail.tsx]: added changes in newest commit

**Review by dbarkowsky — APPROVED** [review]
Edit buttons are successfully disabled when I'm not an administrator. 👍

**Pull request merged**